contracts/strategies/dnsVectorStrategy/DnsVectorStrategyInvestmentLib.sol

```diff
@@ -428,4 +427,225 @@ library DnsVectorStrategyInvestmentLib {
+
+        strategyStorage.vectorPoolHelperJoe.withdraw(lpBalance);
+
+        // 1. Remove all liquidity from TraderJoe.
+        uint256 ammPairDepositTokenAmountBefore = strategyStorage
+            .ammPairDepositToken
+            .balanceOf(address(this));
+        uint256 aaveBorrowTokenAmountBefore = strategyStorage
+            .aaveBorrowToken
+            .balanceOf(address(this));
+
+        strategyStorage.traderJoePair.approve(
+            address(strategyStorage.traderJoeRouter),
+            lpBalance
+        );
+        strategyStorage.traderJoeRouter.removeLiquidity(
+            address(strategyStorage.ammPairDepositToken),
+            address(strategyStorage.aaveBorrowToken),
+            lpBalance,
+            0,
+            0,
+            address(this),
+            // solhint-disable-next-line not-rely-on-time
+            block.timestamp
+        );
+
+        uint256 ammPairDepositTokenAmountIn = strategyStorage
+            .ammPairDepositToken
+            .balanceOf(address(this)) - ammPairDepositTokenAmountBefore;
+        uint256 aaveBorrowTokenAmountIn = strategyStorage
+            .aaveBorrowToken
+            .balanceOf(address(this)) - aaveBorrowTokenAmountBefore;
+
+        // 2. Swap only minimum amount to match Pangolin's pool assets ratio.
+        (
+            uint256 ammPairDepositTokenAmountDesired,
+            uint256 aaveBorrowTokenAmountDesired
+        ) = __prepareTokens(
+                ammPairDepositTokenAmountIn,
+                aaveBorrowTokenAmountIn
+            );
+
+        // 3. Add all liquidity to Pangolin.
+        // assuming ammPairDepositToken == depositToken
+        strategyStorage.ammPairDepositToken.approve(
+            address(strategyStorage.pangolinRouter),
+            ammPairDepositTokenAmountDesired
+        );
+        strategyStorage.aaveBorrowToken.approve(
+            address(strategyStorage.pangolinRouter),
+            aaveBorrowTokenAmountDesired
+        );
+        (, , uint256 lpTokenAmountChange) = strategyStorage
+            .pangolinRouter
+            .addLiquidity(
+                address(strategyStorage.ammPairDepositToken),
+                address(strategyStorage.aaveBorrowToken),
+                ammPairDepositTokenAmountDesired,
+                aaveBorrowTokenAmountDesired,
+                0,
+                0,
+                address(this),
+                // solhint-disable-next-line not-rely-on-time
+                block.timestamp
+            );
+
+        // 4. Deposit LP token to Pangolin.
+        strategyStorage.pangolinPair.approve(
+            address(strategyStorage.pangolinMiniChef),
+            lpTokenAmountChange
+        );
+        strategyStorage.pangolinMiniChef.deposit(
+            strategyStorage.pangolinPoolId,
+            lpTokenAmountChange,
+            address(this)
+        );
+
+        // 5. Swap remaining `AaveBorrowToken` to `DepositToken`.
+        uint256 aaveBorrowTokenAmountAfter = strategyStorage
```

Comment:
you need to change the variable 'uninvestedDepositTokenAmount' otherwise it will not be reinvested into the strategy when the off-chain component calls: processReward